File: workers/java/io/temporal/omes/apps/helloworld/HelloWorldApp.java
package io.temporal.omes.apps.helloworld;

import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.omes.harness.Harness;
import io.temporal.omes.harness.HarnessClients;
import io.temporal.omes.harness.ProjectHarness;
import io.temporal.omes.harness.WorkerHarness;
import io.temporal.worker.Worker;

public final class HelloWorldApp {
  public static final Harness.App APP =
      new Harness.App(
          HelloWorldApp::configureWorker,
          HarnessClients::defaultClientFactory,
          new ProjectHarness.ProjectHandlers(HelloWorldApp::executeProjectIteration));

  private HelloWorldApp() {}

  private static void configureWorker(
      WorkflowClient client, Worker worker, WorkerHarness.WorkerContext context) {
    worker.registerWorkflowImplementationTypes(HelloWorldWorkflowImpl.class);
  }

  private static void executeProjectIteration(
      WorkflowClient client, ProjectHarness.ProjectExecuteContext context) {
    HelloWorldWorkflow workflow =
        client.newWorkflowStub(
            HelloWorldWorkflow.class,
            WorkflowOptions.newBuilder()
                .setWorkflowId(String.format("%s-%d", context.run.executionId, context.iteration))
                .setTaskQueue(context.taskQueue)
                .build());
    String result = workflow.run("World");
    System.out.println(result);
  }
}

File: workers/dotnet/Apps/HelloWorld/HelloWorldApp.cs
using Temporalio.Client;
using Temporalio.Omes.Projects.Harness;
using Temporalio.Worker;

namespace Temporalio.Omes.Apps.HelloWorld;

public static class HelloWorldApp
{
    public static Temporalio.Omes.Projects.Harness.App Create() =>
        new(
            Worker: CreateWorker,
            ClientFactory: ClientHelpers.DefaultClientFactory,
            Project: new ProjectHandlers(Execute: ExecuteProjectIterationAsync));

    private static TemporalWorker CreateWorker(ITemporalClient client, WorkerContext context)
    {
        var workerOptions = context.WorkerOptions;
        workerOptions.AddWorkflow<HelloWorldWorkflow>();
        return new TemporalWorker(client, workerOptions);
    }

    private static async Task ExecuteProjectIterationAsync(ITemporalClient client, ProjectExecuteContext context)
    {
        var handle = await client.StartWorkflowAsync(
            (HelloWorldWorkflow workflow) => workflow.RunAsync("World"),
            new WorkflowOptions(
                id: $"{context.Run.ExecutionId}-{context.Iteration}",
                taskQueue: context.TaskQueue));
        var result = await handle.GetResultAsync<string>();
        Console.WriteLine(result);
    }
}
